apps/backend/src/app/modules/users/users.service.ts

```diff
@@ -281,6 +281,46 @@ export class UsersService {
     await this.db.user.update({ where: { id: userID }, data: updateInput });
   }
 
+  async delete(userID: number) {
+    const user = await this.db.user.findUnique({
+      where: { id: userID }
+    });
+
+    if (!user) throw new NotFoundException('Could not find the user');
+    if (Bitflags.has(user.roles, Role.DELETED)) return;
+
+    const updateInputToClean: Prisma.UserUpdateInput = {
+      roles: Role.DELETED,
+      steamID: null,
+      alias: 'Deleted User',
+      avatar: null,
+      country: null,
+      profile: {
+        update: { bio: '', socials: {} }
+      },
+      mapFavorites: { deleteMany: {} },
+      mapLibraryEntries: { deleteMany: {} },
+      activities: { deleteMany: {} },
+      follows: { deleteMany: {} },
+      followers: { deleteMany: {} },
+      notifications: { deleteMany: {} },
+      runSessions: { deleteMany: {} }
```

Comment:
These shouldn't be needed - our Prisma schema should be configured Postgres to cascade delete all of these. If any aren't being deleted, add an `onDelete: Cascade` to the relation.

Reply:
I know that they cascade delete, and I wanted to use it. The issue is that we don't delete the user

Reply:
OH, I'm sorry, I'm such an idiot.